components/custom_status/custom_status_modal.tsx

```diff
@@ -49,20 +54,43 @@ const defaultCustomStatusSuggestions: DefaultUserCustomStatus[] = [
 ];
 
 const CustomStatusModal: React.FC<Props> = (props: Props) => {
-    const getCustomStatus = makeGetCustomStatus();
     const dispatch = useDispatch();
     const currentCustomStatus = useSelector((state: GlobalState) => getCustomStatus(state)) || {};
     const recentCustomStatuses = useSelector((state: GlobalState) => getRecentCustomStatuses(state));
     const customStatusControlRef = useRef<HTMLDivElement>(null);
     const {formatMessage} = useIntl();
     const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
     const [text, setText] = useState<string>(currentCustomStatus.text || '');
-    const [emoji, setEmoji] = useState<string>(currentCustomStatus.emoji);
-    const [expiry, setExpiry] = useState<string>('four-hours');
+    const [emoji, setEmoji] = useState<string>(currentCustomStatus.emoji || '');
+    const [expiry, setExpiry] = useState<Duration>(currentCustomStatus.duration || Duration.FOUR_HOURS);
     const isStatusSet = emoji || text;
     const isCurrentCustomStatusSet = currentCustomStatus.text || currentCustomStatus.emoji;
     const firstTimeModalOpened = useSelector((state: GlobalState) => showStatusDropdownPulsatingDot(state));
-    const showDateAndTimeField = isStatusSet && expiry === 'date-and-time';
+    const showDateAndTimeField = isStatusSet && expiry === Duration.DATE_AND_TIME;
+
+    const currentUserId = useSelector(getCurrentUserId);
+    const userTimezone = useSelector((state: GlobalState) => getUserTimezone(state, currentUserId));
+    const enableTimezone = useSelector(areTimezonesEnabledAndSupported);
+
+    let currentTime: Date;
+    let timezone: string | undefined;
+    if (enableTimezone) {
+        if (userTimezone.useAutomaticTimezone) {
+            currentTime = getCurrentDateAndTimeForTimezone(userTimezone.automaticTimezone);
+            timezone = userTimezone.automaticTimezone;
+        } else {
+            currentTime = getCurrentDateAndTimeForTimezone(userTimezone.manualTimezone);
+            timezone = userTimezone.manualTimezone;
+        }
+    } else {
+        currentTime = new Date();
+    }
```

Comment:
I think this can still be improved, but here's a suggestion:

```suggestion
    let currentTime = new Date();
    let timezone: string | undefined;
    if (enableTimezone) {
        timezone = userTimezone.manualTimezone;
        if (userTimezone.useAutomaticTimezone) {
            timezone = userTimezone.automaticTimezone;
        }
        currentTime = getCurrentDateAndTimeForTimezone(timezone);
    }
```